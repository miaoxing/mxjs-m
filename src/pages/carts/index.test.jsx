import {render, waitFor, fireEvent, waitForElementToBeRemoved} from '@testing-library/react';
import $, {Ret} from 'miaoxing';
import {createPromise, bootstrap, setUrl, resetUrl} from '@mxjs/test';
import * as TaroTest from '@tarojs/taro';
import Taro from '@tarojs/taro';
import {reset} from 'use-uid';
import Index from './index';
import {init} from 'taro-test';
import {createProduct} from '../../tests/utils';

init();

bootstrap();
let didShow;
// eslint-disable-next-line no-import-assign
TaroTest.useDidShow = (fn) => {
  didShow = fn;
};

describe('Index', () => {
  beforeEach(() => {
    reset();
    setUrl('/carts');
  });

  afterEach(() => {
    resetUrl();
  });

  test('empty', async () => {
    const promise = createPromise();

    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [],
        }),
      }));

    const {container, getByText} = render(<Index />);
    didShow();

    await waitFor(() => {
      expect(getByText('购物车空空如也')).not.toBeNull();
    });

    expect(container).toMatchSnapshot();
    expect($.http).toMatchSnapshot();
  });

  test('list', async () => {
    const product = createProduct();

    const promise = createPromise();

    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[0].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[0],
              product: product,
              createOrder: Ret.suc(),
            },
            {
              id: 2,
              productId: product.id,
              skuId: product.skus[1].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[1],
              product: product,
              createOrder: Ret.err('该商品已失效'),
            },
            {
              id: 3,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.err('该规格已失效'),
            },
          ],
          selected: [1, 3],
        }),
      }));

    const {container, getByText} = render(<Index />);
    didShow();

    await waitFor(() => {
      // 合计：￥81
      expect(getByText('81')).not.toBeNull();
    });

    expect(container).toMatchSnapshot();
    expect($.http).toMatchSnapshot();
  });

  test('select', async () => {
    const product = createProduct();

    const promise = createPromise();

    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[0].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[0],
              product: product,
              createOrder: Ret.suc(),
            },
            {
              id: 2,
              productId: product.id,
              skuId: product.skus[1].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[1],
              product: product,
              createOrder: Ret.err('该商品已失效'),
            },
            {
              id: 3,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 9,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [1, 3],
        }),
      }));

    const {container, getByText, findByText} = render(<Index />);
    didShow();

    const checkout = await findByText('结算（2）');
    const amount = getByText('合计：');

    const checkboxes = container.querySelectorAll('taro-checkbox-core');
    const checkboxGroups = container.querySelectorAll('taro-checkbox-group-core');
    expect(checkboxes.length).toBe(4);

    // 默认选中1，3
    expect(checkboxes[0].checked).toBeTruthy();
    expect(checkboxes[0].disabled).toBeUndefined();

    expect(checkboxes[1].checked).toBeUndefined();
    expect(checkboxes[1].disabled).toBeTruthy();

    expect(checkboxes[2].checked).toBeTruthy();
    expect(checkboxes[2].disabled).toBeUndefined();

    await waitFor(() => expect(checkout.textContent).toBe('结算（2）'));
    expect(amount.textContent).toBe('合计：￥' + (9 * 9 + 12 * 9));

    // 取消选择1
    fireEvent(checkboxGroups[0], new CustomEvent('change', {
      detail: {
        value: [3],
      },
    }));
    expect(checkboxes[0].checked).toBeUndefined();
    expect(checkboxes[1].checked).toBeUndefined();
    expect(checkboxes[2].checked).toBeTruthy();

    await waitFor(() => expect(checkout.textContent).toBe('结算（1）'));
    expect(amount.textContent).toBe('合计：￥' + (12 * 9));

    // 全选
    fireEvent.change(checkboxGroups[1]);
    expect(checkboxes[0].checked).toBeTruthy();
    expect(checkboxes[1].checked).toBeUndefined();
    expect(checkboxes[2].checked).toBeTruthy();

    await waitFor(() => expect(checkout.textContent).toBe('结算（2）'));
    expect(amount.textContent).toBe('合计：￥' + (9 * 9 + 12 * 9));

    // 取消全选
    fireEvent.change(checkboxGroups[1]);
    expect(checkboxes[0].checked).toBeUndefined();
    expect(checkboxes[1].checked).toBeUndefined();
    expect(checkboxes[2].checked).toBeUndefined();

    await waitFor(() => expect(checkout.textContent).toBe('结算（0）'));
    expect(amount.textContent).toBe('合计：￥0');

    // 选择1
    fireEvent(checkboxGroups[0], new CustomEvent('change', {
      detail: {
        value: [1],
      },
    }));
    expect(checkboxes[0].checked).toBeTruthy();
    expect(checkboxes[1].checked).toBeUndefined();
    expect(checkboxes[2].checked).toBeUndefined();

    await waitFor(() => expect(checkout.textContent).toBe('结算（1）'));
    expect(amount.textContent).toBe('合计：￥' + (9 * 9));

    // 提交
    Taro.navigateTo = jest.fn();
    fireEvent.click(checkout);
    await waitFor(() => expect(Taro.navigateTo).toBeCalled());

    expect($.http).toMatchSnapshot();
  });

  test('quantity', async () => {
    const product = createProduct();

    const promise = createPromise();
    const promise2 = createPromise();
    const promise3 = createPromise();

    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [1],
        }),
      }))
      .mockImplementationOnce(() => promise2.resolve({
        ret: Ret.suc(),
      }))
      .mockImplementationOnce(() => promise3.resolve({
        ret: Ret.suc(),
      }));

    const {container, getByText} = render(<Index />);
    didShow();

    await waitFor(() => {
      expect(getByText('结算（1）')).not.toBeNull();
    });

    const amount = getByText('60');

    const stepper = container.querySelector('.mx-stepper-input');

    const minus = container.querySelector('.mx-stepper-minus');
    fireEvent.click(minus);
    expect(stepper.value).toBe(4);
    expect(amount.textContent).toBe('￥48');

    const plus = container.querySelector('.mx-stepper-plus');
    fireEvent.click(plus);
    expect(stepper.value).toBe(5);
    expect(amount.textContent).toBe('￥60');

    expect($.http).toMatchSnapshot();
  });

  test('delete', async () => {
    const product = createProduct();

    const promise = createPromise();
    const promise2 = createPromise();
    $.http = jest.fn()
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [1],
        }),
      }))
      .mockImplementationOnce(() => promise2.resolve({
        ret: Ret.suc(),
      }));

    const {getByText, queryByText} = render(<Index />);
    didShow();

    await waitFor(() => {
      expect(getByText('结算（1）')).not.toBeNull();
    });

    const remove = getByText('移除');

    fireEvent.click(remove);

    await waitForElementToBeRemoved(() => queryByText('移除'));

    expect(getByText('购物车空空如也')).not.toBeNull();
    expect(getByText('结算（0）')).not.toBeNull();

    const amount = getByText('合计：');
    expect(amount.textContent).toBe('合计：￥0');

    expect($.http).toMatchSnapshot();
  });

  test('checkout', async () => {
    const product = createProduct();
    $.http = jest.fn()
      .mockResolvedValueOnce({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [],
        }),
      });
    $.err = jest.fn();

    const {container, findByText} = render(<Index />);
    didShow();

    await findByText(product.name);
    expect($.http).toMatchSnapshot();

    const checkout = await findByText('结算（0）');
    fireEvent.click(checkout);
    expect($.err).toBeCalledWith('请选择要结算的商品');

    fireEvent.change(container.querySelectorAll('taro-checkbox-group-core')[1]);
    expect(checkout.textContent).toBe('结算（1）');

    Taro.navigateTo = jest.fn();
    fireEvent.click(checkout);
    await waitFor(() => expect(Taro.navigateTo).toBeCalled());
  });

  test('update sku', async () => {
    const product = createProduct();
    $.http = jest.fn()
      .mockResolvedValueOnce({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[2],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [1],
        }),
      });

    const {getByText, findByText} = render(<Index />);
    didShow();

    await findByText(product.name);

    const amount = getByText('合计：');
    expect(amount.textContent).toBe('合计：￥' + (5 * 12));

    $.http.mockResolvedValueOnce({
      ret: Ret.suc({
        data: product,
      }),
    });
    const skuValues = getByText('M；红色');
    fireEvent.click(skuValues);

    const colorBlue = await findByText('蓝色');
    fireEvent.click(colorBlue);

    $.http.mockResolvedValueOnce({ret: Ret.suc()});
    $.http.mockResolvedValueOnce({
      ret: Ret.suc({
        data: [
          {
            id: 1,
            productId: product.id,
            skuId: product.skus[3].id,
            quantity: 5,
            changedPrice: null,
            addedPrice: '9',
            configs: {},
            // updated
            updatedAt: Date.now(),
            sku: product.skus[3],
            product: product,
            createOrder: Ret.suc(),
          },
        ],
        selected: [1],
      }),
    });
    fireEvent.click(getByText('确 定'));

    await waitFor(() => expect(amount.textContent).toBe('合计：￥' + (5 * 14)));
  });

  test('product deleted', async () => {
    const product = createProduct();
    product.deletedAt = Date.now();
    $.http = jest.fn()
      .mockResolvedValueOnce({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: product.skus[2].id,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              product: product,
              createOrder: Ret.err('该商品已失效'),
            },
          ],
          selected: [],
        }),
      });
    $.err = jest.fn();

    const {container, findByText, queryByText} = render(<Index />);
    didShow();

    await findByText(product.name);
    expect($.http).toMatchSnapshot();

    // 不显示规格，价格，步进器
    expect(queryByText('红色')).toBeNull();
    expect(container.querySelector('.mx-stepper')).toBeNull();
    expect(queryByText(product.skus[2].price)).toBeNull();

    // 显示错误提示
    expect(queryByText('该商品已失效')).not.toBeNull();
  });

  test('sku deleted', async () => {
    const product = createProduct();
    $.http = jest.fn()
      .mockResolvedValueOnce({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: null,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: null,
              product: product,
              createOrder: Ret.err('该规格已失效'),
            },
          ],
          selected: [],
        }),
      });

    const {findByText, getByText, queryByText} = render(<Index />);
    didShow();

    await findByText(product.name);

    $.http.mockResolvedValueOnce({
      ret: Ret.suc({
        data: product,
      }),
    });
    const reSelect = getByText('重选');
    fireEvent.click(reSelect);

    $.http.mockResolvedValueOnce({ret: Ret.suc()});
    $.http.mockResolvedValueOnce({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              productId: product.id,
              skuId: 4,
              quantity: 5,
              changedPrice: null,
              addedPrice: '9',
              configs: {},
              updatedAt: '2021-06-28 16:45:59',
              sku: product.skus[3],
              product: product,
              createOrder: Ret.suc(),
            },
          ],
          selected: [],
        }),
      });
    fireEvent.click(await findByText('蓝色'));
    fireEvent.click(await findByText('M'));
    fireEvent.click(getByText('确 定'));

    await waitForElementToBeRemoved(() => queryByText('重选'));

    expect($.http).toMatchSnapshot();
  });
});
