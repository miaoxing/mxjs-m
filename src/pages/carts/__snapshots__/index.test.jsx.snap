// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index checkout 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Index delete 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
    Array [
      Object {
        "method": "delete",
        "url": "/api/carts/1",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`Index empty 1`] = `
<div>
  <taro-view-core
    class="mx-page css-0"
    style="--mx-colors-gray-50: #f9fafb; --mx-colors-gray-100: #f3f4f6; --mx-colors-gray-200: #e5e7eb; --mx-colors-gray-300: #d1d5db; --mx-colors-gray-400: #9ca3af; --mx-colors-gray-500: #6b7280; --mx-colors-gray-600: #4b5563; --mx-colors-gray-700: #374151; --mx-colors-gray-800: #1f2937; --mx-colors-gray-900: #111827; --mx-space-0: 0; --mx-space-1: 4; --mx-space-2: 8; --mx-space-3: 12; --mx-space-4: 16; --mx-space-5: 20; --mx-space-6: 24; --mx-space-7: 28; --mx-space-8: 32; --mx-space-9: 36; --mx-space-10: 40; --mx-space-11: 44; --mx-space-12: 48; --mx-space-14: 56; --mx-space-16: 64; --mx-space-20: 80; --mx-space-24: 96; --mx-space-28: 112; --mx-space-32: 128; --mx-space-36: 144; --mx-space-40: 160; --mx-space-44: 176; --mx-space-48: 192; --mx-space-52: 208; --mx-space-56: 224; --mx-space-60: 240; --mx-space-64: 256; --mx-space-72: 288; --mx-space-80: 320; --mx-space-96: 384; --mx-fontSizes-xs: 12; --mx-fontSizes-sm: 14; --mx-fontSizes-base: 16; --mx-fontSizes-lg: 18; --mx-fontSizes-xl: 20; --mx-fontSizes-2xl: 24; --mx-fontSizes-3xl: 30; --mx-fontSizes-4xl: 36; --mx-fontSizes-5xl: 48; --mx-fontSizes-6xl: 60; --mx-fontSizes-7xl: 72; --mx-fontSizes-8xl: 96; --mx-fontSizes-9xl: 128;"
  >
    <taro-checkbox-group-core />
    <taro-view-core
      style="flex-direction: column; color: rgb(107, 114, 128); justify-content: center; align-items: center; display: flex;"
    >
      <taro-view-core
        class="mx-empty-image"
        style="width: 128px; height: 82px;"
      />
      <taro-view-core>
        购物车空空如也
      </taro-view-core>
      <taro-view-core>
        <taro-navigator-core
          url="/products"
        >
          <taro-button-core
            class="mx-btn mx-btn-primary"
          >
            去逛逛
          </taro-button-core>
        </taro-navigator-core>
      </taro-view-core>
    </taro-view-core>
    <taro-view-core
      class="mx-footer-bar __mx-footer-bar-1"
      style="flex-direction: row; align-items: center; justify-content: space-between; display: flex;"
    >
      <taro-checkbox-group-core>
        <taro-label-core
          style="color: rgb(107, 114, 128); line-height: 1; flex-direction: row; align-items: center; display: flex;"
        >
          <taro-checkbox-core />
          全选
        </taro-label-core>
      </taro-checkbox-group-core>
      <taro-view-core
        style="flex-direction: row; align-items: center; display: flex;"
      >
        <taro-view-core
          style="color: rgb(107, 114, 128);"
        >
          合计：
          <taro-text-core
            primary500="true"
          >
            <taro-text-core>
              ￥
            </taro-text-core>
            0
          </taro-text-core>
        </taro-view-core>
        <taro-view-core
          class="mx-btn-theme mx-btn-theme-pill"
        >
          <taro-button-core
            class="mx-btn mx-btn-primary"
          >
            结算（
            0
            ）
          </taro-button-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </taro-view-core>
</div>
`;

exports[`Index empty 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`Index list 1`] = `
<div>
  <taro-view-core
    class="mx-page css-0"
    style="--mx-colors-gray-50: #f9fafb; --mx-colors-gray-100: #f3f4f6; --mx-colors-gray-200: #e5e7eb; --mx-colors-gray-300: #d1d5db; --mx-colors-gray-400: #9ca3af; --mx-colors-gray-500: #6b7280; --mx-colors-gray-600: #4b5563; --mx-colors-gray-700: #374151; --mx-colors-gray-800: #1f2937; --mx-colors-gray-900: #111827; --mx-space-0: 0; --mx-space-1: 4; --mx-space-2: 8; --mx-space-3: 12; --mx-space-4: 16; --mx-space-5: 20; --mx-space-6: 24; --mx-space-7: 28; --mx-space-8: 32; --mx-space-9: 36; --mx-space-10: 40; --mx-space-11: 44; --mx-space-12: 48; --mx-space-14: 56; --mx-space-16: 64; --mx-space-20: 80; --mx-space-24: 96; --mx-space-28: 112; --mx-space-32: 128; --mx-space-36: 144; --mx-space-40: 160; --mx-space-44: 176; --mx-space-48: 192; --mx-space-52: 208; --mx-space-56: 224; --mx-space-60: 240; --mx-space-64: 256; --mx-space-72: 288; --mx-space-80: 320; --mx-space-96: 384; --mx-fontSizes-xs: 12; --mx-fontSizes-sm: 14; --mx-fontSizes-base: 16; --mx-fontSizes-lg: 18; --mx-fontSizes-xl: 20; --mx-fontSizes-2xl: 24; --mx-fontSizes-3xl: 30; --mx-fontSizes-4xl: 36; --mx-fontSizes-5xl: 48; --mx-fontSizes-6xl: 60; --mx-fontSizes-7xl: 72; --mx-fontSizes-8xl: 96; --mx-fontSizes-9xl: 128;"
  >
    <taro-checkbox-group-core>
      <taro-view-core
        class="mx-card"
      >
        <taro-view-core
          class="mx-list"
        >
          <taro-view-core
            style="overflow: hidden;"
          >
            <taro-view-core
              class="mx-swipe-action-content"
              style="transform: translate3d(0px, 0, 0);"
            >
              <taro-view-core>
                <taro-view-core
                  class="mx-list-item"
                >
                  <taro-view-core
                    class="mx-list-col"
                    style="flex: 0 0px;"
                  >
                    <taro-view-core
                      style="flex-direction: row; align-items: center; display: flex;"
                    >
                      <taro-checkbox-core
                        checked="true"
                        value="1"
                      />
                      <taro-navigator-core
                        style="border-radius: 8rpx; overflow: hidden;"
                        url="/products/show?id=1"
                      >
                        <taro-image-core
                          mode="aspectFit"
                          src="https://dev.test.com/uploads/1/210610/002439539409.png"
                        />
                      </taro-navigator-core>
                    </taro-view-core>
                  </taro-view-core>
                  <taro-view-core
                    class="mx-list-col"
                    style="flex-direction: column; justify-content: space-between; display: flex;"
                  >
                    <taro-navigator-core
                      hoverclass="none"
                      truncate2="true"
                      url="/products/show?id=1"
                    >
                      alias commodi ipsam
                    </taro-navigator-core>
                    <taro-view-core>
                      <taro-view-core
                        style="display: inline-block; color: rgb(107, 114, 128); background-color: rgb(243, 244, 246); border-radius: 4rpx;"
                      >
                        S；红色
                        <taro-text-core
                          class="mxi mxi-chevron-down"
                        />
                      </taro-view-core>
                    </taro-view-core>
                    <taro-view-core
                      style="flex-direction: row; align-items: center; justify-content: space-between; display: flex;"
                    >
                      <taro-text-core
                        primary500="true"
                      >
                        <taro-text-core>
                          ￥
                        </taro-text-core>
                        9
                      </taro-text-core>
                      <taro-view-core>
                        <taro-view-core
                          class="mx-stepper"
                        >
                          <taro-button-core
                            class="mx-stepper-minus"
                          />
                          <taro-input-core
                            class="mx-stepper-input"
                            value="9"
                          />
                          <taro-button-core
                            class="mx-stepper-plus"
                            disabled="true"
                          />
                        </taro-view-core>
                      </taro-view-core>
                    </taro-view-core>
                  </taro-view-core>
                </taro-view-core>
              </taro-view-core>
              <taro-view-core
                class="mx-swipe-action-options __mx-swipe-action-options-2"
              >
                <taro-view-core
                  style="color: rgb(255, 255, 255); background-color: rgb(239, 68, 68); flex-direction: row; justify-content: center; align-items: center; display: flex;"
                >
                  移除
                </taro-view-core>
              </taro-view-core>
            </taro-view-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
      <taro-view-core
        class="mx-card"
      >
        <taro-view-core
          class="mx-list"
        >
          <taro-view-core
            style="overflow: hidden;"
          >
            <taro-view-core
              class="mx-swipe-action-content"
              style="transform: translate3d(0px, 0, 0);"
            >
              <taro-view-core>
                <taro-view-core
                  class="mx-list-item"
                >
                  <taro-view-core
                    class="mx-list-col"
                    style="flex: 0 0px;"
                  >
                    <taro-view-core
                      style="flex-direction: row; align-items: center; display: flex;"
                    >
                      <taro-checkbox-core
                        disabled="true"
                        value="2"
                      />
                      <taro-navigator-core
                        style="border-radius: 8rpx; overflow: hidden;"
                        url="/products/show?id=1"
                      >
                        <taro-image-core
                          mode="aspectFit"
                          src="https://dev.test.com/uploads/1/210610/002439539409.png"
                        />
                      </taro-navigator-core>
                    </taro-view-core>
                  </taro-view-core>
                  <taro-view-core
                    class="mx-list-col"
                    style="flex-direction: column; justify-content: space-between; display: flex;"
                  >
                    <taro-navigator-core
                      hoverclass="none"
                      truncate2="true"
                      url="/products/show?id=1"
                    >
                      alias commodi ipsam
                    </taro-navigator-core>
                    <taro-view-core>
                      <taro-view-core
                        style="display: inline-block; color: rgb(107, 114, 128); background-color: rgb(243, 244, 246); border-radius: 4rpx;"
                      >
                        S；蓝色
                        <taro-text-core
                          class="mxi mxi-chevron-down"
                        />
                      </taro-view-core>
                    </taro-view-core>
                    <taro-view-core
                      style="flex-direction: row; align-items: center; justify-content: space-between; display: flex;"
                    >
                      <taro-text-core
                        primary500="true"
                      >
                        <taro-text-core>
                          ￥
                        </taro-text-core>
                        11
                      </taro-text-core>
                      <taro-view-core>
                        <taro-view-core
                          class="mx-stepper"
                        >
                          <taro-button-core
                            class="mx-stepper-minus"
                          />
                          <taro-input-core
                            class="mx-stepper-input"
                            value="9"
                          />
                          <taro-button-core
                            class="mx-stepper-plus"
                            disabled="true"
                          />
                        </taro-view-core>
                      </taro-view-core>
                    </taro-view-core>
                    <taro-view-core
                      style="color: rgb(239, 68, 68);"
                    >
                      该商品已失效
                    </taro-view-core>
                  </taro-view-core>
                </taro-view-core>
              </taro-view-core>
              <taro-view-core
                class="mx-swipe-action-options __mx-swipe-action-options-3"
              >
                <taro-view-core
                  style="color: rgb(255, 255, 255); background-color: rgb(239, 68, 68); flex-direction: row; justify-content: center; align-items: center; display: flex;"
                >
                  移除
                </taro-view-core>
              </taro-view-core>
            </taro-view-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
      <taro-view-core
        class="mx-card"
      >
        <taro-view-core
          class="mx-list"
        >
          <taro-view-core
            style="overflow: hidden;"
          >
            <taro-view-core
              class="mx-swipe-action-content"
              style="transform: translate3d(0px, 0, 0);"
            >
              <taro-view-core>
                <taro-view-core
                  class="mx-list-item"
                >
                  <taro-view-core
                    class="mx-list-col"
                    style="flex: 0 0px;"
                  >
                    <taro-view-core
                      style="flex-direction: row; align-items: center; display: flex;"
                    >
                      <taro-checkbox-core
                        disabled="true"
                        value="3"
                      />
                      <taro-navigator-core
                        style="border-radius: 8rpx; overflow: hidden;"
                        url="/products/show?id=1"
                      >
                        <taro-image-core
                          mode="aspectFit"
                          src="https://dev.test.com/uploads/1/210610/002439539409.png"
                        />
                      </taro-navigator-core>
                    </taro-view-core>
                  </taro-view-core>
                  <taro-view-core
                    class="mx-list-col"
                    style="flex-direction: column; justify-content: space-between; display: flex;"
                  >
                    <taro-navigator-core
                      hoverclass="none"
                      truncate2="true"
                      url="/products/show?id=1"
                    >
                      alias commodi ipsam
                    </taro-navigator-core>
                    <taro-view-core>
                      <taro-view-core
                        style="display: inline-block; color: rgb(107, 114, 128); background-color: rgb(243, 244, 246); border-radius: 4rpx;"
                      >
                        M；红色
                        <taro-text-core
                          class="mxi mxi-chevron-down"
                        />
                      </taro-view-core>
                    </taro-view-core>
                    <taro-view-core
                      style="flex-direction: row; align-items: center; justify-content: space-between; display: flex;"
                    >
                      <taro-text-core
                        primary500="true"
                      >
                        <taro-text-core>
                          ￥
                        </taro-text-core>
                        12
                      </taro-text-core>
                      <taro-view-core>
                        <taro-view-core
                          class="mx-stepper"
                        >
                          <taro-button-core
                            class="mx-stepper-minus"
                          />
                          <taro-input-core
                            class="mx-stepper-input"
                            value="9"
                          />
                          <taro-button-core
                            class="mx-stepper-plus"
                            disabled="true"
                          />
                        </taro-view-core>
                      </taro-view-core>
                    </taro-view-core>
                    <taro-view-core
                      style="color: rgb(239, 68, 68);"
                    >
                      该规格已失效
                    </taro-view-core>
                  </taro-view-core>
                </taro-view-core>
              </taro-view-core>
              <taro-view-core
                class="mx-swipe-action-options __mx-swipe-action-options-4"
              >
                <taro-view-core
                  style="color: rgb(255, 255, 255); background-color: rgb(239, 68, 68); flex-direction: row; justify-content: center; align-items: center; display: flex;"
                >
                  移除
                </taro-view-core>
              </taro-view-core>
            </taro-view-core>
          </taro-view-core>
        </taro-view-core>
      </taro-view-core>
    </taro-checkbox-group-core>
    <taro-view-core
      class="mx-footer-bar __mx-footer-bar-1"
      style="flex-direction: row; align-items: center; justify-content: space-between; display: flex;"
    >
      <taro-checkbox-group-core>
        <taro-label-core
          style="color: rgb(107, 114, 128); line-height: 1; flex-direction: row; align-items: center; display: flex;"
        >
          <taro-checkbox-core
            checked="true"
          />
          全选
        </taro-label-core>
      </taro-checkbox-group-core>
      <taro-view-core
        style="flex-direction: row; align-items: center; display: flex;"
      >
        <taro-view-core
          style="color: rgb(107, 114, 128);"
        >
          合计：
          <taro-text-core
            primary500="true"
          >
            <taro-text-core>
              ￥
            </taro-text-core>
            81
          </taro-text-core>
        </taro-view-core>
        <taro-view-core
          class="mx-btn-theme mx-btn-theme-pill"
        >
          <taro-button-core
            class="mx-btn mx-btn-primary"
          >
            结算（
            1
            ）
          </taro-button-core>
        </taro-view-core>
      </taro-view-core>
    </taro-view-core>
  </taro-view-core>
</div>
`;

exports[`Index list 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`Index product deleted 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`Index quantity 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
    Array [
      Object {
        "data": Object {
          "quantity": 4,
        },
        "method": "put",
        "url": "/api/carts/1/quantity",
      },
    ],
    Array [
      Object {
        "data": Object {
          "quantity": 5,
        },
        "method": "put",
        "url": "/api/carts/1/quantity",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;

exports[`Index select 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
    Array [
      Object {
        "data": Object {
          "ids": Array [
            3,
          ],
        },
        "method": "put",
        "url": "/api/carts/selected",
      },
    ],
    Array [
      Object {
        "data": Object {
          "ids": Array [
            1,
            3,
          ],
        },
        "method": "put",
        "url": "/api/carts/selected",
      },
    ],
    Array [
      Object {
        "data": Object {
          "ids": Array [],
        },
        "method": "put",
        "url": "/api/carts/selected",
      },
    ],
    Array [
      Object {
        "data": Object {
          "ids": Array [
            1,
          ],
        },
        "method": "put",
        "url": "/api/carts/selected",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Index sku deleted 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
    Array [
      Object {
        "method": "get",
        "url": "/api/products/1",
      },
    ],
    Array [
      Object {
        "data": Object {
          "quantity": 5,
          "skuId": 4,
        },
        "method": "patch",
        "url": "/api/carts/1",
      },
    ],
    Array [
      Object {
        "url": "/api/carts",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
